app/core/pokemon-entity.ts

```diff
@@ -535,6 +535,28 @@ export class PokemonEntity extends Schema implements IPokemonEntity {
       )
     }
   }
+  
+  addPermanentStats(
+    stat: { [K in keyof IPokemon]:
+      IPokemon[K] extends number ? K : never
+      }[keyof IPokemon],
+    value: number
+  ) {
+    if(this.isGhostOpponent) return
+    this.refToBoardPokemon[stat] += value
```

Comment:
I suggest refactoring the `applyStat` function so that it takes a `permanent` boolean flag, this way we can reuse all the value-checks / rounding / min / max of all the existing stat buffs methods in PokemonEntity

Reply:
Yeah that's a much better idea. Should've thought about that first.

Reply:
Oh I think my change isn't what you wanted. Let me see about doing the full refactor

Reply:
ok should be good now